invenio_users_resources/services/permissions.py

```diff
@@ -65,6 +66,7 @@ class GroupsPermissionPolicy(BasePermissionPolicy):
     _can_any = [
         GroupsEnabled("group"),
         SystemProcess(),
+        UserManager,
```

Comment:
I have the feeling permissions might be a bit more complicated than the rest. Specially, if we don't want administration users to mess up with super admin permissions.
Check what we did, https://github.com/ubiquitypress/invenio-users-resources/blob/up-v8.3.0-1/invenio_users_resources/services/permissions.py, it's not perfect but it works 😅

Reply:
I tried implementing your permission approach in the PR on top of this one: https://github.com/inveniosoftware/invenio-users-resources/pull/187/files#diff-c9d321f56f73e3aaf314b698eeca0dfc8894624de5adf93f558943fb5754013c

but many unrelated tests started failing. So, I reverted the change and kept the generators so we can discuss it together.
We can update the permissions once this PR is merged while implementing the user role updates.